grails-app/services/com/lucastex/grails/fileuploader/FileUploaderService.groovy

```diff
@@ -341,32 +341,45 @@ class FileUploaderService {
             return null
         }
         //Create temp directory
+        UrlValidator urlValidator = new UrlValidator()
         String tempDirectory = "./web-app/temp/${System.currentTimeMillis()}/"
         new File(tempDirectory).mkdirs()
 
         //create file
         def tempFile = "${tempDirectory}/${ufileInstance.name}" // No need to append extension. name field already have that.
         def destFile = new File(tempFile)
-        def sourceFile = new File(ufileInstance.path)
         if (!destFile.exists()) {
             destFile.createNewFile()
         }
 
-        FileChannel source = null
-        FileChannel destination = null
+        String sourceFilePath = ufileInstance.path
+        if (urlValidator.isValid(sourceFilePath)) {

```

Comment:
Actually you have to check for the `type` field in `UFile` domain to check if it is a cloud based ufile. Url validator is a good option but that needs to be in secondary check.